src/net.c

```diff
@@ -3033,6 +3033,21 @@ handle_response(coap_context_t *context, coap_session_t *session,
   if (rcvd->type != COAP_MESSAGE_ACK)
     coap_cancel_all_messages(context, session, rcvd->token, rcvd->token_length);
 
+  /* Check for message duplication */
+  if (COAP_PROTO_NOT_RELIABLE(session->proto)) {
+    if (rcvd->type == COAP_MESSAGE_CON) {
+      if (rcvd->mid == session->last_con_mid)
+        /* Duplicate response */
```

Comment:
Especially because there is a comment here, I would like to include the return in a block. Same for rcvd->mid==session->last_ack_mid below.